### PR TITLE
[AutoScheduler][FIX] Fix exception handling in measure.py

### PR DESCRIPTION
**Pull request by shingjan**
This PR intends to fix one exception handling in `auto_scheduler/measure.py`.

@junrushao1994 @vinx13 @tqchen

### CODE REVIEW

File: python/tvm/auto_scheduler/measure.py [hunk 1]
```diff
@@ -718,11 +709,18 @@ def local_builder_build(inputs, timeout, n_parallel, build_func="default", verbo
...
-            assert res.status == StatusKind.TIMEOUT
+        elif res.status == StatusKind.TIMEOUT:
             if verbose >= 1:
                 print(".T", end="", flush=True)  # Build timeout
             results.append(BuildResult(None, [], MeasureErrorNo.BUILD_TIMEOUT, None, timeout))
+        elif res.status == StatusKind.EXCEPTION:
+            if verbose >= 1:
+                print(".E", end="", flush=True)  # Build error
+            results.append(
+                BuildResult(None, [], MeasureErrorNo.COMPILE_HOST, make_traceback_info(), timeout)
```
Comment (junrushao1994) [inline python/tvm/auto_scheduler/measure.py:720]: `make_traceback_info` on the host side won't reflect the actual error that happens on the worker. Let's use the exception propagated from `res.value` instead. Don't forget the stringify the exception :-)

**Comment by junrushao1994** [comment]
Let's get the patch done and merged quickly to prevent potential flakiness on mainline

**Review by comaniac — APPROVED** [review]
LGTM

File: python/tvm/auto_scheduler/measure.py (hunk 1 above)
Comment (shingjan) [reply python/tvm/auto_scheduler/measure.py]: done

**Review by vinx13 — APPROVED** [review]

**Review by junrushao1994 — APPROVED** [review]
Let's get it merged quickly as I am seeing several failed flaky tests on i386

**Pull request merged**